components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/internal/action/PreUpdateProfileActionExecutor.java

```diff
@@ -0,0 +1,304 @@
+                            IdentityContext.getThreadLocalIdentityContext().getTenantDomain());
+
+            handleActionExecutionStatus(actionExecutionStatus);
+        } catch (ActionExecutionException e) {
+            throw new UserActionExecutionServerException(UserActionError.PRE_UPDATE_PROFILE_ACTION_SERVER_ERROR,
+                    "Error while executing pre update profile action.", e);
+        }
+    }
+
+    /**
+     * Triggers the execution of pre update profile extension
+     *
+     * @param user                                             SCIMUserObject reference that updates
+     * @param userClaimsExcludingMultiValuedClaimsToBeModified Collection of new claims and existing claims that
+     *                                                         updates in the profile, excluding multi-valued claims
+     *                                                         with list of values separated with
+     *                                                         multi attribute separator
+     * @param userClaimsExcludingMultiValuedClaimsToBeDeleted  Collection of existing claims that deletes from the
+     *                                                         profile, excluding multi-valued claims with list of
+     *                                                         values separated with multi attribute separator
+     * @param simpleMultiValuedClaimsToBeAdded                 Collection of claims that updates with a list of values
+     *                                                         that are added or updated in existing value set
+     * @param simpleMultiValuedClaimsToBeRemoved               Collection of claims that includes a list of values
+     *                                                         that are removed from existing value set
+     * @param existingClaimsOfUser                             The existing claims of the user
+     * @throws UserStoreException If an error occurs while executing the action
+     */
+    public void execute(User user, Map<String, String> userClaimsExcludingMultiValuedClaimsToBeModified,
+                        Map<String, String> userClaimsExcludingMultiValuedClaimsToBeDeleted,
+                        Map<String, List<String>> simpleMultiValuedClaimsToBeAdded,
+                        Map<String, List<String>> simpleMultiValuedClaimsToBeRemoved,
+                        Map<String, String> existingClaimsOfUser)
+            throws UserStoreException {
+
+        ActionExecutorService actionExecutorService = SCIMCommonComponentHolder.getActionExecutorService();
+
+        if (!actionExecutorService.isExecutionEnabled(ActionType.PRE_UPDATE_PROFILE)) {
+            return;
+        }
+
+        LOG.debug("Executing pre update profile action for user: " + user.getId());
+
+        try {
+            Map<String, String> multiValuedClaimsToModify = getSimpleMultiValuedClaimsToModify(existingClaimsOfUser,
```

Comment:
Shouldn't we check the whether MultiAttributeSeparator is present in claim values.

Reply:
Didn't get your question.
This basically converts the the values that should be either added or values that should be either removed from the value set of a multi valued claim and then combine the final value set for a one single string using the multi attribute value separator.
This same logic is executed in the user store level before storing these claims